src/jira_client.ts

```diff
@@ -0,0 +1,381 @@
+  // tslint:disable-next-line:no-any
+  fetchApi: typeof fetch = (nodeFetch as any) as typeof fetch
+) =>
+  tryCatch(() => {
+    return fetchApi(`${baseUrl}/rest/api/2/search`, {
+      method: "POST",
+
+      headers: {
+        Accept: "application/json",
+        Authorization: `Basic ${Buffer.from(`${jiraEmail}:${token}`).toString(
+          "base64"
+        )}`,
+        "Content-Type": "application/json"
+      },
+
+      body: JSON.stringify(bodyData)
+    });
+  }, toError).chain<SearchJiraIssueResponse>(_ => {
+    if (_.status >= 500) {
+      return fromLeft(new Error("Jira API returns an error"));
+    }
+    if (_.status === 401) {
+      return fromLeft(new Error("Jira secrets misconfiguration"));
+    }
+    if (_.status === 400) {
+      return fromLeft(new Error("Wrong Jira JQL"));
+    }
+    if (_.status !== 200) {
+      return fromLeft(new Error("Unknown status code response error"));
+    }
+    return tryCatch(() => _.json(), toError).chain(responseBody => {
+      return fromEither(
+        SearchJiraIssueResponse.decode(responseBody).mapLeft(errors => {
+          return toError(readableReport(errors));
+        })
+      );
+    });
+  });
+
+export interface IJiraAPIClient {
+  readonly createJiraIssue: (
+    title: NonEmptyString,
+    description: NonEmptyString,
+    serviceId: NonEmptyString,
+    labels?: ReadonlyArray<NonEmptyString>
+  ) => TaskEither<Error, CreateJiraIssueResponse>;
+  readonly createJiraIssueComment: (
+    issueId: NonEmptyString,
+    comment: NonEmptyString
+  ) => TaskEither<Error, CreateJiraCommentIssueResponse>;
+  readonly getServiceJiraIssuesByStatus: (params: {
+    readonly serviceId: ServiceId;
+    readonly status: NonEmptyString;
+  }) => TaskEither<Error, SearchJiraIssueResponse>;
+  readonly searchServiceJiraIssue: (params: {
+    readonly serviceId: ServiceId;
+  }) => TaskEither<Error, SearchJiraIssueResponse>;
+  readonly applyJiraIssueTransition: (
+    issueId: NonEmptyString,
+    transitionId: NonEmptyString,
+    newComment?: NonEmptyString
+  ) => TaskEither<Error, "OK">;
+  readonly deleteJiraIssue: (
+    issueId: NonEmptyString
+  ) => TaskEither<Error, "OK">;
+}
+
+export function JiraAPIClient(
+  baseUrl: string,
+  jiraEmail: string,
+  token: string,
+  boardId: string,
+  // tslint:disable-next-line:no-any
+  fetchApi: typeof fetch = (nodeFetch as any) as typeof fetch
+): IJiraAPIClient {
+  const createJiraIssue = (
+    title: NonEmptyString,
+    description: NonEmptyString,
+    serviceId: NonEmptyString,
+    labels?: ReadonlyArray<NonEmptyString>
+  ) => {
+    return tryCatch(
+      () =>
+        fetchApi(`${baseUrl}/rest/api/2/issue`, {
+          method: "POST",
+
+          headers: {
+            Accept: "application/json",
+            Authorization: `Basic ${Buffer.from(
+              `${jiraEmail}:${token}`
+            ).toString("base64")}`,
+            "Content-Type": "application/json"
+          },
```

Comment:
`headers` are always the same into all methods inside `JiraAPIClient`. What if we create an unexported constant inside `JiraAPIClient` and we use that value all over the fetch?
es. 
```
const jiraHeaders = {
    Accept: "application/json",
    Authorization: `Basic ${Buffer.from(`${jiraEmail}:${token}`).toString(
      "base64"
    )}`,
    "Content-Type": "application/json"
  };
```

Reply:
Fix with d52f2df